File: app/src/main/java/com/kickstarter/services/apirequests/ResetPasswordBody.java
package com.kickstarter.services.apirequests;

import android.os.Parcelable;

import com.kickstarter.libs.qualifiers.AutoGson;

import auto.parcel.AutoParcel;

@AutoGson
@AutoParcel
public abstract class ResetPasswordBody implements Parcelable {
  public abstract String email();

  @AutoParcel.Builder
  public abstract static class Builder {
    public abstract Builder email(String __);
    public abstract ResetPasswordBody build();
  }

  public static Builder builder() {
    return new AutoParcel_ResetPasswordBody.Builder();
  }

  public abstract Builder toBuilder();
}

File: app/src/main/java/com/kickstarter/services/apirequests/ResetPasswordBody.kt
package com.kickstarter.services.apirequests

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
class ResetPasswordBody private constructor(
    private val email: String
) : Parcelable {
    fun email() = this.email

    @Parcelize
    data class Builder(
        private var email: String = ""
    ) : Parcelable {
        fun email(email: String) = apply { this.email = email }
        fun build() = ResetPasswordBody(
            email = email
        )
    }

    fun toBuilder() = Builder(
        email = email
    )

    override fun equals(obj: Any?): Boolean {
        var equals = super.equals(obj)
        if (obj is ResetPasswordBody) {
            equals = email() == obj.email()
        }
        return equals
    }

    companion object {
        @JvmStatic
        fun builder() = Builder()
    }
}
